sdks/python/apache_beam/transforms/util.py

```diff
@@ -959,52 +1053,19 @@ class ReshufflePerKey(PTransform):
   """
   def expand(self, pcoll):
     windowing_saved = pcoll.windowing
-    if windowing_saved.is_default():
-      # In this (common) case we can use a trivial trigger driver
-      # and avoid the (expensive) window param.
-      globally_windowed = window.GlobalWindows.windowed_value(None)
-      MIN_TIMESTAMP = window.MIN_TIMESTAMP
-
-      def reify_timestamps(element, timestamp=DoFn.TimestampParam):
-        key, value = element
-        if timestamp == MIN_TIMESTAMP:
-          timestamp = None
-        return key, (value, timestamp)
-
-      def restore_timestamps(element):
-        key, values = element
-        return [
-            globally_windowed.with_value((key, value)) if timestamp is None else
-            window.GlobalWindows.windowed_value((key, value), timestamp)
-            for (value, timestamp) in values
-        ]
-
-      if is_compat_version_prior_to(pcoll.pipeline.options,
-                                    RESHUFFLE_TYPEHINT_BREAKING_CHANGE_VERSION):
-        pre_gbk_map = Map(reify_timestamps).with_output_types(Any)
-      else:
-        pre_gbk_map = Map(reify_timestamps).with_input_types(
-            tuple[K, V]).with_output_types(
-                tuple[K, tuple[V, Optional[Timestamp]]])
-    else:
-
-      # typing: All conditional function variants must have identical signatures
-      def reify_timestamps(  # type: ignore[misc]
-          element, timestamp=DoFn.TimestampParam, window=DoFn.WindowParam):
-        key, value = element
-        # Transport the window as part of the value and restore it later.
-        return key, windowed_value.WindowedValue(value, timestamp, [window])
-
-      def restore_timestamps(element):
-        key, windowed_values = element
-        return [wv.with_value((key, wv.value)) for wv in windowed_values]
-
-      if is_compat_version_prior_to(pcoll.pipeline.options,
-                                    RESHUFFLE_TYPEHINT_BREAKING_CHANGE_VERSION):
-        pre_gbk_map = Map(reify_timestamps).with_output_types(Any)
-      else:
-        pre_gbk_map = Map(reify_timestamps).with_input_types(
-            tuple[K, V]).with_output_types(tuple[K, TypedWindowedValue[V]])
+    is_default_windowing = windowing_saved.is_default()
+    should_keep_paneinfo = not is_compat_version_prior_to(
```

Comment:
Instead of adding flags that increase the cyclomatic complexity of `expand`, prefer to immediately fork `expand` to another version (which we will not modify ever again, really). See https://github.com/apache/beam/blob/4fe34db0f42633aac1045591b14fdaabd664a43c/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Reshuffle.java#L87